apis/fluentbit/v1alpha2/plugins/common_types.go

```diff
@@ -1,18 +1,26 @@
 package plugins
 
-import "github.com/fluent/fluent-operator/apis/fluentbit/v1alpha2/plugins/params"
+import (
+	"github.com/fluent/fluent-operator/apis/fluentbit/v1alpha2/plugins/params"
+)
 
 // +kubebuilder:object:generate:=true
 
 type CommonParams struct {
 
 	// Alias for the plugin
 	Alias string `json:"alias,omitempty"`
+	// Integer value to set the maximum number of retries allowed. N must be >= 1
```

Comment:
This note should be more detailed and accurate.

Reply:
> > @wanjunlei please review again, I think it should be good now :) The only thing missing is a sort of check if the supplied value for Retry_Limit is an acceptable value. Right now users could supply values like `retryLimit: "I am Batman"` and fluent-operator would accept it and write this broken config. I could add a check that checks if "false" is supplied or any valid integer. However I wouldn't know what to do in case the supplied value is illegal. Just log it so users can see the error when watching the operators logs and not write it to any output's config or what should happen in case an illegal value is found?
> 
> That is good. A simple way to check the value is used [// +kubebuilder:validation:Pattern](https://book.kubebuilder.io/reference/markers/crd-validation.html#crd-validation) annotation, you can try it.

I tried that in the updated version please review. I am not entirely sure if `+kubebuilder:validation:XIntOrString=true` is needed or not though.

Reply:
> I improved the description.
> 
> > > @wanjunlei please review again, I think it should be good now :) The only thing missing is a sort of check if the supplied value for Retry_Limit is an acceptable value. Right now users could supply values like `retryLimit: "I am Batman"` and fluent-operator would accept it and write this broken config. I could add a check that checks if "false" is supplied or any valid integer. However I wouldn't know what to do in case the supplied value is illegal. Just log it so users can see the error when watching the operators logs and not write it to any output's config or what should happen in case an illegal value is found?
> > 
> > 
> > That is good. A simple way to check the value is used [// +kubebuilder:validation:Pattern](https://book.kubebuilder.io/reference/markers/crd-validation.html#crd-validation) annotation, you can try it.
> 
> I tried that in the updated version please review. I am not entirely sure if `+kubebuilder:validation:XIntOrString=true` is needed or not though.

I think  `+kubebuilder:validation:XIntOrString=true` is not needed because it duplicates the `+kubebuilder:validation:Pattern` . 

In addition, there are two small problems here. One is that the pattern `^((false)|([1-9]+[0-9]*))$` does not support `no_retries`, this value is used to disable retry, and it better support `False` too. 

Another is that the default value should be `1`.

<img width="624" alt="1667957372990" src="https://user-images.githubusercontent.com/53003665/200714389-e5978cdd-60d7-47aa-8849-f6222b05cdce.png">

Reply:
That's good, thanks for your pr.